modin/experimental/core/execution/native/implementations/hdk_on_native/base_worker.py

```diff
@@ -239,6 +237,41 @@ def cast_to_compatible_types(table):
 
         return table
 
+    @staticmethod
+    def _convert_field(field, field_type):
```

Comment:
Question about consistency. Why introduce `_convert_field` if some cases have not been moved to it?

It seems to me that the code would be easier to read in the following form:
```python
for i, field in enumerate(schema):
  if pa.types.is_dictionary(field.type):
    if pa.types.is_null(value_type):
      [update flags + create field]
    elif pa.types.is_string(value_type):
      [update flags + create field]
  if pa.types.is_date(field_type):
    [update flags + create field]
  elif pa.types.is_unsigned_integer(field_type):
    [update flags + create field]
  new_schema = new_schema.set(i, new_field)
```

Reply:
`_convert_field` introduced to convert both field types and dictionary value types. What cases have not been moved?

Reply:
`is_null` and `is_string` cases

Reply:
These cases are only related to dictionaries. As mentioned in comments, a string column is created for dictionary of null. String dictionaries are converted to string columns only if HDK version is < 1.8.

Reply:
> It seems to me that the code would be easier to read in the following form:

```python
for i, field in enumerate(schema):
  if pa.types.is_dictionary(field.type):
    if pa.types.is_null(value_type):
      [update flags + create field]
    elif pa.types.is_string(value_type):
      [update flags + create field]
  if pa.types.is_date(field_type):
    [update flags + create field]
  elif pa.types.is_unsigned_integer(field_type):
    [update flags + create field]
  new_schema = new_schema.set(i, new_field)
```

@AndreyPavlenko what about this?

Reply:
This code handles only null and string dictionary fields, but ignores any other. Else statement is missing after `elif pa.types.is_string`.  In case of dictionary, the field should be converted to the value type, except of null and string, which have  a different conversion logic. Non dictionary fields are converted only in case of date and uint. Dictionary fields of type date and uint are also converted in the same way. I.e. dictionary fields are converted to the value type and, additionally, for the date and string types, the `_convert_field()` logic is applied either.